test/unit/test-mysqlDao.js

```diff
@@ -1898,4 +1898,112 @@ describe('MysqlDao', function() {
         });
     });
 
+    describe('join', function() {
+
+        before(done => {
+            databaseSetup(err => {
+                if (err) return done(err);
+
+                const dummyUsers = [
+                    {id: 1, firstName: 'John', lastName: 'Doe', email: 'john.doe@gmail.com'},
+                    {id: 2, firstName: 'Jane', lastName: 'Doe', email: 'jane.doe@gmail.com'},
+                    {id: 3, firstName: 'James', lastName: 'Bond', email: 'lax4life-007@gmail.com'}
+                ];
+
+                const dummyTopSecretInfo = [
+                    {secretAgentCode: 1, handlerCode: null, intel: "It's gonna rain."},
+                    {secretAgentCode: 1, handlerCode: null, intel: "The sky is blue."},
+                    {secretAgentCode: 99, handlerCode: 2, intel: "Hello world."}
+                ];
+
+                // Create 500 pieces of high quality intel for James Bond.
+                _.range(500).reverse().map(i => dummyTopSecretInfo.push(
+                    {secretAgentCode: 3, handlerCode: null, intel: `${i} shaken-not-stirred martinis on the wall, ${i} shaken-not-stirred martinis...`}
+                ));
+
+                return Promise.all([
+                    userDao.createBulk(dummyUsers),
+                    topSecretInfoDao.createBulk(dummyTopSecretInfo)
+                ])
+                    .then(() => done())
+                    .catch(done)
+
+            })
+        });
+
+        it('Should join all relevant target entities to an array of source entities', function() {
+
+            return userDao.getAll({})
+                .then(users => {
+                    return topSecretInfoDao.leftJoin(users, 'secretAgentCode')
```

Comment:
Now that I'm thinking....this might be a rightJoin?  It's users leftJoin topSecretInfo but topSecretInfo rightJoin users.